ha_cluster_exporter.go

```diff
@@ -84,53 +100,92 @@ func loglevel(opt string) {
 	case "debug":
 		log.SetLevel(log.DebugLevel)
 	default:
-		log.Warnln("Unrecognized log level, default to `info` level")
+		log.Warnln("Unrecognized minimum log level; using 'info' as default")
 	}
 }
 
-var (
-	clock      Clock = &SystemClock{}
-	portNumber       = flag.String("port", "9002", "The port number to listen on for HTTP requests.")
-	address          = flag.String("address", "0.0.0.0", "The address to listen on for HTTP requests.")
-	logLevel         = flag.String("level", "info", "The level of logs to log")
-)
+var clock Clock = &SystemClock{}
+
+func init() {
+	config.SetConfigName("ha_cluster_exporter")
+	config.AddConfigPath("./")
+	config.AddConfigPath("$HOME/.config/")
+	config.AddConfigPath("/etc/")
+
+	flag.String("port", "9002", "The port number to listen on for HTTP requests")
+	flag.String("address", "0.0.0.0", "The address to listen on for HTTP requests")
+	flag.String("log-level", "info", "The minimum logging level; levels are, in ascending order: debug, info, warn, error")
+	flag.String("crm-mon-path", "/usr/sbin/crm_mon", "path to crm_mon executable")
+	flag.String("cibadmin-path", "/usr/sbin/cibadmin", "path to cibadmin executable")
+	flag.String("corosync-cfgtoolpath-path", "/usr/sbin/corosync-cfgtool", "path to corosync-cfgtool executable")
+	flag.String("corosync-quorumtool-path", "/usr/sbin/corosync-quorumtool", "path to corosync-quorumtool executable")
+	flag.String("sbd-path", "/usr/sbin/sbd", "path to sbd executable")
+	flag.String("sbd-config-path", "/etc/sysconfig/sbd", "path to sbd configuration")
+	flag.String("drbdsetup-path", "/usr/sbin/drbdsetup", "path to drbdsetup executable")
+
+	err := config.BindPFlags(flag.CommandLine)
+	if err != nil {
+		log.Errorf("Could not bind config to CLI flags: %v", err)
+	}
+}
 
 func main() {
-	// read cli option and setup initial stat
+	var err error
```

Comment:
why this duplicata? we can delete this line and just add on line 137 `:=`

Reply:
this was actually done on purpose: since the main has gotten very long, declaring `err` at the top avoids errors in case the first assignment is moved around.